packages/app-mobile/root.tsx

```diff
@@ -890,6 +897,37 @@ class AppComponent extends React.Component {
 			</View>
 		);
 	}
+
+	private async navigate(url: string) {
+		if (!url) return;
+		let parsedUrl;
+		try {
+			parsedUrl = new URL(url);
+		} catch (_) {
+			// invalid URL
```

Comment:
Please throw an error - there's not reason for the url to be invalid since we provide it.